command/genesis/utils.go

```diff
@@ -98,43 +105,61 @@ func parsePremineInfo(premineInfoRaw string) (*premineInfo, error) {
 	return &premineInfo{address: address, balance: amount}, nil
 }
 
+// matchValidatorDataDirs uses regexp to match validator directories which must have the last
+// three digits: non-digit, hyphen(-), digit.
+// Found paths are appended to validatorDataPaths.paths
+func (v *validatorDataPaths) matchValidatorDataDirs(path string, info fs.FileInfo, _ error) error {
+	if info.IsDir() && strings.HasPrefix(info.Name(), filepath.Base(v.pathPrefix)) {
+		// match paths that have a number as the last digit /tmp/data-1, /tmp/test-data-2, etc.
+		// matches any string that has last 3 chars: not-digit, - , digit
+		match, _ := regexp.MatchString(`^.*\D-\d{1}$`, strings.TrimSpace(path))
```

Comment:
Two things:
- `^.*\D-\d{1}$` with this regexp, I haven't managed to capture e.g.test-chain, so my question is should we even match non-digit here, since seems the result is the same as running this regexp `^.*-\d{1}$`? Can you give a counter-example?
- consider putting this regexp as const?